### PR TITLE
Allow user to download files in Hapmap format (VCF bulk)

### PR DESCRIPTION
**Pull request by Jiu9Shen**
Allow users download filtered vcf files in Hapmap format.

There are three modified files includes/vcf_filter.form.inc; includes/vcf_filter.trpdownload.inc; vcf_filter.module.

One new file (includes/vcf_filter.trpdownload.hapmap.inc) with two functions:
1. vcf_filter_hapmap_generate_file: handles the download of hapmap format file
1. vcf_filter_convert_VCF_to_Hapmap: handles the conversion from vcf format to hapmap format

### CODE REVIEW

File: includes/vcf_filter.trpdownload.hapmap.inc [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+ */
+
+function vcf_filter_convert_VCF_to_Hapmap ($in_vcf_filename, $out_hapmap_filename){
+
+  //create output file
+  $handle_out = fopen($out_hapmap_filename,'w') or die ("\nCannot create file: $out_hapmap_filename .\n");
+
+  // read vcf file by lines
+  foreach (new SplFileObject($in_vcf_filename) as $line)
+  {
```
Comment (carolyncaron) [inline includes/vcf_filter.trpdownload.hapmap.inc]: I know this sounds really nit-picky, but Drupal standards require that opening curly brackets be placed immediately after the function/loop/if statement, and not on its own line. Closing curly brackets remain on their own line, as you've already done! See here for examples: https://www.drupal.org/docs/develop/standards/coding-standards#functdecl

Can you please fix this throughout your code to maintain consistency?

File: includes/vcf_filter.trpdownload.hapmap.inc [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  }
+  // Half way point...
+  tripal_set_job_progress($job_id, 50);
+  // FORMAT.
+  //===============
+
+  //------------------------------------------------------------------
+  // new script starts here, above codes are copied with minor changes
+  // the extension of filename can to be '.hmp' 
+  //$hapmap_filename = preg_replace('/\.txt$/', '/\.hmp/', $variables['filename']);
```
Comment (carolyncaron) [inline includes/vcf_filter.trpdownload.hapmap.inc]: Please remove commented code

File: includes/vcf_filter.trpdownload.hapmap.inc [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  $filtered_vcf = vcf_filter_filter_file($vcf_file->file_path, $variables['q']);
+  if (!$filtered_vcf) {
+    exit(3);
+  }
+  // Half way point...
+  tripal_set_job_progress($job_id, 50);
+  // FORMAT.
+  //===============
+
+  //------------------------------------------------------------------
```
Comment (carolyncaron) [inline includes/vcf_filter.trpdownload.hapmap.inc]: I don't think this comment block is very informative

File: includes/vcf_filter.form.inc [hunk 4]
```diff
@@ -280,6 +285,7 @@ function vcf_filter_form_submit($form, &$form_state) {
     'matrix' => '/filter_vcf/SNP-Matrix',
     'qual_matrix' => '/filter_vcf/Quality-Matrix',
     'bgzipped' => '/filter_vcf/Bgzipped',
+	'hapmap' => '/filter_vcf/Hapmap',
```
[the remark this answers is not in the document]
Comment (Jiu9Shen) [reply includes/vcf_filter.form.inc]: Still, tab space issue, I think I need to change tabs to spaces for all files I've ever touched...

**Comment by laceysanderson** [comment]
Apparently this is already live on portal despite not being merged in ;-)

**Pull request merged**